reader.py

```diff
@@ -0,0 +1,24 @@
+import http.client
+import json
+
+# Purpose of class: to make Http request to specified URI using GET request to retrieve data
+class HttpRequestResponse:
+    def __init__(self, link, request):
+      self.link = link
+      self.request = request
+
+    #Establish connection and get data
+    def establish_connection(self, link, request, start):
+        print('----------Start----------')
+        print("Start Time: " + str(start))
+        conn = http.client.HTTPSConnection(self.link)
+        headers = {'Content-type': 'application/json'}
+        conn.request("GET", self.request, headers=headers)
```

Comment:
Whenever we make a request to a remote server, it is a good practice to have a try and except. Find out about how to implement try and except for request function in requests module. Also implement if the exception occurred.

Reply:
try and catch added for request connection and json decode to handle null value